tests/integration/test_services_auto_tagging_copilot_feedback.py

```
@@ -0,0 +1,1217 @@
+        rec = svc.suggest_tags(f, existing_tags=["already"], top_n=5)
+        suggested = [s.tag for s in rec.suggestions]
+        assert "already" not in suggested
+
+
+# ---------------------------------------------------------------------------
+# PreviewEngine (copilot rules)
+# ---------------------------------------------------------------------------
+
+
+class TestPreviewEngine:
+    def _make_rule_set(self, rules=None):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_make_rule_set`

(ANN202)